src/lavinmq/shovel/shovel.cr

```diff
@@ -316,18 +320,28 @@ module LavinMQ
           end
           @log.error(exception: ex) { ex.message }
           @error = ex.message
-          sleep @reconnect_delay.seconds
+          exponential_reconnect_delay
         rescue ex
           break if terminated?
           @state = State::Error
           @log.error(exception: ex) { ex.message }
           @error = ex.message
-          sleep @reconnect_delay.seconds
+          exponential_reconnect_delay
         end
       ensure
         terminate
       end
 
+      def exponential_reconnect_delay
+        @retries += 1
+        if @retries > 5
+          sleep @delay.seconds
+          @delay = [@delay * 2, 300].min unless @delay > 300
+        else
+          sleep @delay.seconds
+        end
```

Comment:
```suggestion
        sleep @delay.seconds
        if @retries > 5
          @delay = [@delay * 2, 300].min
        end
```
I guess it can simplified to this. And maybe change `5` and `300` to constants with describing names?